src/ts/types.ts

```diff
@@ -1,11 +1,54 @@
-export interface ComponentProps {
+export type IntrinsicElementType =
```

Comment:
it feels odd that we need to define this set - does react not have something that defines these already?

Reply:
Yes, but they define WAAAAY more elements (some of which we shouldn't really be using, e.g. it would be mad to set a `Column` to be `component="svg"` or any other SVG element for that matter). I've just chosen a set of elements we'll probably use commonly, and these can be added to when we need.